### PR TITLE
Implemented policy monitoring add operation

### PR DESCRIPTION
**Pull request by milanperera**

### CODE REVIEW

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java [hunk 1]
```diff
@@ -247,13 +251,25 @@ public void addMonitoringOperation(List<Device> devices) throws PolicyCompliance
             String msg = "Error occurred from monitoring dao.";
             log.error(msg, e);
             throw new PolicyComplianceException(msg, e);
+        } catch (OperationManagementException e) {
+            String msg = "Error occurred while adding monitoring operation to devices";
```
Comment (prabathabey) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java:255]: Avoid log and throw unless specifically needed

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java [hunk 2]
```diff
@@ -247,13 +251,25 @@ public void addMonitoringOperation(List<Device> devices) throws PolicyCompliance
...
-    private void addMonitoringOperationsToDatabase(List<Device> devices) throws PolicyComplianceException {
+    private void addMonitoringOperationsToDatabase(List<Device> devices)
+            throws PolicyComplianceException, OperationManagementException {
 
+        List<DeviceIdentifier> deviceIdentifiers = this.getDeviceIdentifiersFromDevices(devices);
+        CommandOperation monitoringOperation = new CommandOperation();
+        monitoringOperation.setEnabled(true);
+        monitoringOperation.setType(Operation.Type.COMMAND);
+        monitoringOperation.setCode(OPERATION_MONITOR);
+        PolicyManagementDataHolder.getInstance().getDeviceManagementService().
```
Comment (DilanUA) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java:271]: Possible null pointer exception in .getDeviceManagementService() . . .?

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java (hunk 1 above)
Comment (milanperera) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java:255]: Removed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/cdbdc86f42b350c24e9d059ff7c23e1f09242df4

File: components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java (hunk 2 above)
Comment (milanperera) [inline components/policy-mgt/org.wso2.carbon.policy.mgt.core/src/main/java/org/wso2/carbon/policy/mgt/core/mgt/impl/MonitoringManagerImpl.java:271]: Added in PR [1].
[1] https://github.com/wso2/carbon-device-mgt/pull/27

**Pull request merged**